examples/e2e/steps/training/model_hp_tuning.py

```diff
@@ -0,0 +1,103 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from typing import Annotated, Any, Dict
+
+import pandas as pd
+from config import MetaConfig
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import RandomizedSearchCV
+
+from zenml import step
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+@step
+def model_hp_tuning(
```

Comment:
Yeah doing the entire tuning in a single step is the only way to do that right now I think. Maybe enabling step calls from within other steps could alleviate this issue since you could then just call your `trainer` step below with different configs from within here directly.